src/lib/appConfig.ts

```diff
@@ -1,31 +1,105 @@
+}
+
+const emptyConfig: Config = {
+  plot: { location: "", sizeGB: 0 },
+  rewardAddress: "",
+  launchOnBoot: true,
+  version: process.env.APP_VERSION || "?",
+  segmentCache: { networkSegmentCount: 0, blockchainSizeGB: 0 },
+}
 
 export const appConfig = {
-  initAppConfig(): void {
-    if(!this.getAppConfig())
-      LocalStorage.set("appConfig", emptyAppConfig)
+  async configDir(): Promise<string> {
+    return (await path.configDir()) + appName
+  },
+  async configFullPath(): Promise<string> {
+    return (await path.configDir()) + appName + "/" + appName + ".cfg"
+  },
+  async init(): Promise<void> {
+    try {
+      await this.read()
+    } catch {
+      // means there were no config file
+      await fs.createDir(await this.configDir()).catch((error) => {
+        if (!error.includes("exists")) {
+          console.error(error)
+        }
+      });
+      await this.write(emptyConfig)
+    }
   },
-  getAppConfig(): AppConfig | void {
-    const appConfig = LocalStorage.getItem("appConfig")
-    if (appConfig) return appConfig as AppConfig
+  validate(config: Config): boolean {
+    const { plot, rewardAddress } = config
+    if (
+      plot.location.length > 0 &&
+      plot.sizeGB > 0 &&
+      rewardAddress.length > 0
+    ) {
+      return true
+    }
+    return false
+  },
+  async remove(): Promise<void> {
+    await fs.removeFile(await this.configFullPath()).catch(console.error)
+  },
+
+  async read(): Promise<Config> {
+    const result = await fs.readTextFile(await this.configFullPath())
+    const config: Config = JSON.parse(result)
+    return config
+  },
+  async write(config: Config): Promise<void> {
+    await fs.createDir(await this.configDir()).catch((error) => {
+      if (!error.includes("exists")) {
+        console.error(error)
+      }
+    });
+    await fs.writeFile({
+        path: await this.configFullPath(),
+        contents: JSON.stringify(config, null, 2)
+      })
+      .catch(console.error)
   },
-  updateAppConfig(
+  async update(
     plot: Plot | null,
-    segmentCache: SegmentCache | null,
-    launchOnBoot: boolean | null,
     rewardAddress: string | null,
-    plottingStarted: boolean | null
-  ): void {
-    const appConfig = this.getAppConfig()
-    if (appConfig) {
-      const newAppConfig = appConfig
-      if (plot) newAppConfig.plot = plot
-      if (segmentCache) newAppConfig.segmentCache = segmentCache
-      if (launchOnBoot != null) newAppConfig.launchOnBoot = launchOnBoot
-      if (rewardAddress) newAppConfig.rewardAddress = rewardAddress
-      if (plottingStarted != null) newAppConfig.plottingStarted = plottingStarted
-      LocalStorage.set("appConfig", newAppConfig)
-    }
+    launchOnBoot: boolean | null,
+    version: string | null,
+    segmentCache: SegmentCache | null,
+  ): Promise<void> {
+    const newAppConfig = await this.read()
+    if (plot) newAppConfig.plot = plot
+    if (launchOnBoot != null) newAppConfig.launchOnBoot = launchOnBoot
```

Comment:
why not `if (launchOnBoot)`?

Reply:
that was a really funny bug. 
Imagine `launchOnBoot` is `false`. 
Then the following line:
`if (launchOnBoot) newAppConfig.launchOnBoot = launchOnBoot`
won't execute, because if statement will evaluate to false :D

Reply:
makes sense, `launchOnBoot !== null` is preferred then. I will adjust eslint rules later